lld/ELF/ScriptParser.cpp

```diff
@@ -1229,6 +1229,9 @@ SymbolAssignment *ScriptParser::readSymbolAssignment(StringRef name) {
 // This is an operator-precedence parser to parse a linker
 // script expression.
 Expr ScriptParser::readExpr() {
+  // Do not try to read expression if an error has already been encountered.
+  if (atEOF())
```

Comment:
I think this will work, although I think it may be more idiomatic to use
```
if (errCount(ctx))
  return 0;
```
That way you don't need the comment to explain that atEOF will return true if there's an error. The return 0 comes from other places where an ErrAlways has occurred, such as
```
      ErrAlways(ctx) << loc << ": division by zero";
      return 0;
```

Reply:
Thank you for your inputs @smithp35.

> although I think it may be more idiomatic to use
>
```
 if (errCount(ctx))
   return 0;
```

I agree that `errCount(ctx)` would work as well, however, I think it is better for code-readability if the behavior of parsing expression is consistent for both the error-case and the actual end-of-file case, and `atEOF()` covers both the cases. Please let me know your thoughts on this.

Reply:
> I've got a small suggestion that means we don't need the comment.

I agree that the comment was a little redundant. I have removed the comment.

Reply:
Thanks for the updates. This looks OK to me. I've added the maintainer MaskRay and MysteryMath, who has been involved in linker script parsing recently.

Reply:
This looks ok to me as well. A few parser functions call `atEOF()` at the beginning, when they expect to consume at least one token.